numpy/core/src/umath/loops.c.src

```diff
@@ -840,28 +845,74 @@ NPY_NO_EXPORT NPY_GCC_OPT_3 void
     UNARY_LOOP_FAST(@type@, @type@, *out = in > 0 ? 1 : (in < 0 ? -1 : 0));
 }
 
+/* Libdivide only supports 32 and 64 bit types
+ * We try to pick the best possible one */
+/**begin repeat1
+ * #kind = t, gen, do#
+ */
+#if NPY_BITSOF_@TYPE@ <= 32
+#define libdivide_@type@_@kind@ libdivide_s32_@kind@
+#else
+#define libdivide_@type@_@kind@ libdivide_s64_@kind@
+#endif
+/**end repeat1**/
+
 NPY_NO_EXPORT void
 @TYPE@_divide(char **args, npy_intp const *dimensions, npy_intp const *steps, void *NPY_UNUSED(func))
 {
-    BINARY_LOOP {
-        const @type@ in1 = *(@type@ *)ip1;
+    BINARY_LOOP_BASE
+
+    if (!is2) {
```

Comment:
I would slightly prefer to make this `steps[1] == 0`, then add a comment that this means the divisor is always identical, which allows a faster algorithm (using libdivide.h).

That is: you trust the compiler to optimize all the unnecessary `+= 0` away.

Reply:
After seeing your suggestions, I got a way to get rid of `+=0` for `ip2` as well as `ip1`. It's a bit more efficient this way.

Reply:
@ganesh-k13 if you replace `BINARY_LOOP_ZERO` with `BINARY_LOOP`, is that actually slower? i would have expected the compiler to optimize it to pretty much the same code.

Reply:
You are right @seberg , the compiler is indeed optimizing. I confirmed with the graph profiler. Although we cannot use `BINARY_LOOP` directly in place of `BINARY_LOOP_ZERO` as it has the initialization code, we need to branch out based on the initialization values, I'll do what Eric suggested below.